rero_ils/modules/items/serializers/collector_ext.py

```diff
@@ -0,0 +1,183 @@
+from rero_ils.modules.documents.utils import title_format_text_head
+from rero_ils.modules.documents.views import work_access_point
+
+from .collector import Collector
+
+
+class CollectorExt(Collector):
+    """collect data for csv."""
+
+    # define chunk size
+    chunk_size = 1000
+    separator = ' | '
+
+    @classmethod
+    def get_documents_by_item_pids(cls, item_pids, language):
+        """Get documents for the given item pid list.
+
+        :param item_pids: item pids.
+        :param language: language.
+        :return list of documents.
+        """
+
+        def _build_doc(data):
+            document_data = {}
+            title = title_format_text_head(
+                data.get('title', []), with_subtitle=True)
+            document_data['document_title'] = title
+            # process contributions
+            creator = []
+            for contribution in data.get('contribution', []):
+                authorized_access_point = \
+                    f'authorized_access_point_{language}'
+                if authorized_access_point in contribution.get('agent'):
+                    creator.append(
+                        f'{contribution["agent"][authorized_access_point]}'
+                        f', {contribution.get("role")}'
+                    )
+            document_data['document_contribution'] = ' ; '.join(creator)
+            document_main_type = []
+            document_sub_type = []
+            for document_type in data.get('type'):
+                document_main_type.append(
+                    document_type.get('main_type'))
+                document_sub_type.append(
+                    document_type.get('subtype', ''))
+            document_data['document_main_type'] = ', '.join(
+                document_main_type)
+            document_data['document_sub_type'] = ', '.join(
+                document_sub_type)
+            # masked
+            document_data['document_masked'] = \
+                'Yes' if data.get('_masked') else 'No'
+            # isbn:
+            document_data['document_isbn'] = cls.separator.join(
+                data.get('isbn', []))
+            # issn:
+            document_data['document_issn'] = cls.separator.join(
+                data.get('issn', []))
+            # document_identifiedBy
+            identifiers = []
+            for identified_by in data.get('identifiedBy', []):
+                identified_type = identified_by.get("type")
+                if identified_type not in ['bf:Isbn', 'bf:Issn']:
+                    identified = (
+                        f'({identified_type}){identified_by.get("value")}')
+                    for additional in [
+                            "source", "status", "qualifier", "note"]:
+                        if more := identified_by.get(additional):
+                            identified = f'{identified}; {additional}: {more}'
+                    identifiers.append(identified)
```

Comment:
We create a "framework" to deal with document identifier to skip all these scripted lines.
```python
identifiers = [
    IdentifierFactory\
        .create_identifier(data)\
        .render(render_class=MyCustomIdentifierRendered())
    for data in self.get('identifiedBy', [])
    if data.get('type') in [IdentifierType.ISBN, IdentifierType.ISSN]
]
```
See `rero_ils/commons/identifiers.py`